R/init.R

```diff
@@ -167,40 +166,53 @@ initDGEobj <- function(primaryAssayData,
     grparent <- itemName
 
     dgeObj <- addItem(dgeObj,
-                      item = rowData,
+                      item     = rowData,
                       itemName = parent,
                       itemType = parent,
-                      funArgs = funArgs,
-                      parent = "")
+                      funArgs  = funArgs,
+                      parent   = "")
 
     dgeObj <- addItem(dgeObj,
-                      item = rowData,
+                      item     = rowData,
                       itemName = itemName,
                       itemType = itemType,
-                      funArgs = funArgs,
-                      parent = parent)
+                      funArgs  = funArgs,
+                      parent   = parent)
 
     # Annotate the level
     dgeObj %<>% setAttributes(list(level = level))
 
-    result <- try({gr <- GenomicRanges::GRanges(rowData)}, silent = TRUE)
-
-    if (class(result) == "try-error") {
-        warning("Couldn't build a GRanges object!")
-    } else {
-        dgeObj <- addItem(dgeObj,
-                          item = gr,
-                          itemName = "granges_orig",
-                          itemType = "granges_orig",
-                          funArgs = funArgs,
-                          parent = paste(grparent, "_orig", sep = ""))
-
-        dgeObj <- addItem(dgeObj,
-                          item = gr,
-                          itemName = "granges",
-                          itemType = "granges",
-                          funArgs = funArgs,
-                          parent = grparent)
+    if (level %in% c("exon", "gene")) {
+        gr_success <- FALSE
```

Comment:
can we use `genomic_ranges_success` instead of `gr_success` to enhance readability and self documentation?

Reply:
yes!  will update